src/components/Quiz/QuizWidget/QuizProgressBar.tsx

```diff
@@ -1,13 +1,22 @@
 import { useCallback } from "react"
 import { Center, ChakraProps, Container } from "@chakra-ui/react"
 
+import type { AnswerChoice, Question } from "@/lib/types"
+
 import { PROGRESS_BAR_GAP } from "@/lib/constants"
 
-import { useQuizWidgetContext } from "./context"
+import type { AnswerStatus } from "./useQuizWidget"
+
+type QuizProgressBarProps = {
+  questions: Question[]
+  answerStatus: AnswerStatus
+  currentQuestionIndex: number
+  userQuizProgress: AnswerChoice[]
+}
 
-export const QuizProgressBar = () => {
+export const QuizProgressBar = (props: QuizProgressBarProps) => {
```

Comment:
@wackerow great question! Lack of consistency here. 

In the `code-convention` docs, I wrote an example where the descructuring occurred at the declaration and not within the component. This allows the props to be visible when hover over the component in use (VSCode Intellisense).

I'll review the components here and apply that convention, because that should be preferred for DX.

(I'm use to doing/seeing both depending on the project I am interacting with!)

Reply:
@wackerow this is now updated with a few of the quiz components.

Now when you hover over the component name, you should see this in VSCode:
![image](https://github.com/ethereum/ethereum-org-website/assets/65234762/01f17517-897f-47df-8b32-51980cc09321)